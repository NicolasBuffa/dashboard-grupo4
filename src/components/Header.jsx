const Header = ({children}) => {
  return (
    <header className="Header">
        {/* boton */}
        {children}
    </header>
  )
}

export default Header