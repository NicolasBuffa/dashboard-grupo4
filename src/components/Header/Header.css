.Header{
    position: fixed;
    height: 78px;
    width: 100%;
    display: flex;
    justify-content: space-between;
    background-color: var(--white_letra);
    padding: 20px;
    align-items: center;
    border-bottom: 2px var(--gray_background) solid;
  }

  @media (min-width: 1024px) {
    
    .Header{
      width: calc(100vw - 296px);
    }
  }

.darkTheme .Header{  
  background-color:var(--gray_background2);
  border-bottom: var(--black_background);
}