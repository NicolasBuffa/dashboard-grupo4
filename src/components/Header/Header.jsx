import MenuButtonMobile from '../SideBar/MenuButton/MenuButtonMobile'
import './Header.css'
import useSideBarController from '../SideBar/useSideBarController'
import { useEffect } from 'react'
const Header = ({children}) => {
  
  let menuButtonClose = document.querySelector('.menuButtonClose');
  let sideBarComponent=document.querySelector('.SideBar');
  
  const {input, ButtonControler} = useSideBarController(sideBarComponent, menuButtonClose)
  useEffect(()=>{

  }, [])
  
  
  return (
    <header className="Header">
        <MenuButtonMobile classBtn={input} btnControler={()=>{ButtonControler()}}/>
        {children}
    </header>
  )
}

export default Header