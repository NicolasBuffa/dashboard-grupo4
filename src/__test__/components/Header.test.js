import Header from '../../components/Header/Header'
import { getByRole, render, screen, act } from "@testing-library/react";
import '@testing-library/jest-dom'

describe('Header tests:', () => {

beforeEach(()=>{const component= render( <Header />)})

    test('Debe renderizarse el Header', () => { 
        const header =screen.getByRole( 'banner')
        expect( header).toBeInTheDocument()
    })
    test('Debe el Header incluir burger-button(SideBar)', () => { 
        const header =screen.getByRole( 'banner')
        // const button= document.querySelector('.menuButton')
        const button= screen.getByRole('button')    
        expect(header).toContainElement(button)
    })    
    test('Debe el Header permanecer siempre visible', () => { 
        const header =screen.getByRole( 'banner');
        expect(header).toMatchSnapshot();
    })
})

describe('Header Childs test:', () =>{
    const TestChild = ()=>{
        return(
            <div>
                <p>TestComponent</p>
            </div>
        )};
        
    test('Debe renderear childrens agregados', () => { 
            const component=render(
                <Header>
                <TestChild />
            </Header>)
        const header =screen.getByRole('banner');
        const testComponent= screen.getByText('TestComponent')
        expect(header).toContainElement(testComponent)
    })
    
})