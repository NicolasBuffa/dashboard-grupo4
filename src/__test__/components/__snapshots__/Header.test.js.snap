// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests: Debe el Header permanecer siempre visible 1`] = `
<header
  class="Header"
  id="Header"
>
  <button
    class="menuButton MenuButtonMobileDesactivate"
  >
    <img
      alt="burgerButton"
      class="imgMenuButton icon"
      src="menu.svg"
    />
  </button>
</header>
`;
